docs/dev/cluster-api.md

```diff
@@ -33,3 +34,96 @@ To build an `openshift-install` binary with Cluster API bundled:
 - Set `export OPENSHIFT_INSTALL_CLUSTER_API=y`
     - Optionally `export SKIP_TERRAFORM=y` if you don't need to use Terraform.
 - Run `./hack/build.sh`, the binary is then produced in `bin/openshift-install`.
+
+### Platform Onboarding
+
+This is a general guide for onboarding a new platform to provision infrastructure with cluster api.
+
+#### CAPI Cloud Provider Controller Dependencies
+
+Providers are vendored to [`cluster-api/providers/<name>`][providerDir]. The README contains
+detailed guidelines on how to vendor a new platform.
+
+#### CAPI Cloud Provider CRDs
+
+To add the CRDs for the CAPI cloud provider to the local control plane, add them to the
+[`/data/data/cluster-api`][crdDir] dir. Typically these CRDs are published as an
```

Comment:
Are the infrastructre-components.yaml files copied from one of the openshift/cluster-api-provider-<mycloud> repos? For example, for gcp, does one copy the file from https://github.com/openshift/cluster-api-provider-gcp/blob/master/openshift/infrastructure-components.yaml into the installer repo? How does one keep this file updated if a new version is published?